com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/resources/ShutdownResource.java

```diff
@@ -0,0 +1,92 @@
+
+import com.sap.cloud.lm.sl.cf.core.flowable.FlowableFacade;
+import com.sap.cloud.lm.sl.cf.core.message.Messages;
+import com.sap.cloud.lm.sl.cf.core.shutdown.model.ShutdownDto;
+
+@Component
+@Produces(MediaType.APPLICATION_JSON)
+public class ShutdownResource {
+
+    @Inject
+    private FlowableFacade flowableFacade;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShutdownResource.class);
+
+    static final String ACTION = "shutdown";
+
+    private static final long DEFAULT_COOLDOWN_TIMEOUT_IN_SECONDS = 300L; // 5 mins
+    private long cooldownTimeoutInSeconds = DEFAULT_COOLDOWN_TIMEOUT_IN_SECONDS;
+
+    @POST
+    public ShutdownDto shutdownFlowableProcessEngine(@HeaderParam("x-cf-applicationid") String appId,
+        @HeaderParam("x-cf-instanceid") String appInstanceId, @HeaderParam("x-cf-instanceindex") String appInstanceIndex,
+        @QueryParam("cooldownTimeoutInSeconds") String cooldownTimeoutInSecondsParam) {
+
+        boolean isFlowableProcessEngineActive = flowableFacade.isProcessEngineActive();
+
+        if (!isFlowableProcessEngineActive) {
+            return new ShutdownDto.Builder().isActive(isFlowableProcessEngineActive)
+                .appId(appId)
+                .appInstanceId(appInstanceId)
+                .appInstanceIndex(appInstanceIndex)
+                .cooldownTimeoutInSeconds(cooldownTimeoutInSeconds)
+                .build();
+        }
+
+        if (cooldownTimeoutInSecondsParam != null) {
+            cooldownTimeoutInSeconds = Long.parseLong(cooldownTimeoutInSecondsParam);
+        }
+
+        CompletableFuture.runAsync(() -> {
+            LOGGER.info(MessageFormat.format(Messages.APP_SHUTDOWN_REQUEST, appId, appInstanceId, appInstanceIndex,
+                cooldownTimeoutInSeconds));
+            flowableFacade.shutdownProcessEngine(cooldownTimeoutInSeconds);
+        })
+            .thenRun(() -> {
+                LOGGER.info(MessageFormat.format(Messages.APP_SHUTDOWNED, appId, appInstanceId, appInstanceIndex,
+                    cooldownTimeoutInSeconds));
+            });
+
+        return new ShutdownDto.Builder().isActive(isFlowableProcessEngineActive)
+            .appId(appId)
+            .appInstanceId(appInstanceId)
+            .appInstanceIndex(appInstanceIndex)
+            .cooldownTimeoutInSeconds(cooldownTimeoutInSeconds)
+            .build();
+    }
+
+    @GET
+    public ShutdownDto getShutdownStatus(@HeaderParam("x-cf-applicationid") String appId,
+        @HeaderParam("x-cf-instanceid") String appInstanceId, @HeaderParam("x-cf-instanceindex") String appInstanceIndex) {
+
+        ShutdownDto shutdownDto = new ShutdownDto.Builder().isActive(flowableFacade.isProcessEngineActive())
+            .appId(appId)
+            .appInstanceId(appInstanceId)
+            .appInstanceIndex(appInstanceIndex)
+            .cooldownTimeoutInSeconds(cooldownTimeoutInSeconds)
+            .build();
+
+        LOGGER.info(
```

Comment:
You could remove the cooldown from logging as it is already reported in the POST request.